cmake/macros.cmake

```diff
@@ -400,6 +400,19 @@ function(SDL_PrintSummary)
       message(STATUS "The library will not be able to create windows on most unix environments.")
       message(STATUS "")
     endif()
+    if(NOT SDL_NO_UNIX_DESKTOP_SANITY_CHECK)
```

Comment:
This needs to go inside the previous `if` block, right?

Also, suggest using `if(NOT SDL_UNIX_CONSOLE_BUILD)` to avoid the double negative and more clearly indicate when you might want this path. - actually maybe wrap the previous check in that, and remove the messages above, since that clearly indicates that they don't want the error checked at all?

Reply:
Indeed, it should have gone in the previous `if` block.

I applied all your suggestions, and also added haiku and riscos the the list of platforms that don't need x11 or wayland.